tests/gtest/avifclaptest.cc

```diff
@@ -129,27 +131,46 @@ constexpr ValidClapPropertyParam kValidClapPropertyTestParams[] = {
      100,
      AVIF_PIXEL_FORMAT_YUV420,
      {99, 1, 99, 1, static_cast<uint32_t>(-1), 2, static_cast<uint32_t>(-1), 2},
-     {0, 0, 99, 99}},
+     {0, 0, 99, 99},
+     false},
+    // pcX = -1/2 + (100 - 1)/2 = 49
+    // pcY = -1/2 + (100 - 1)/2 = 49
+    // leftmost = 49 - (99 - 1)/2 = 0
+    // topmost = 49 - (99 - 1)/2 = 0
+    {100,
+     100,
+     AVIF_PIXEL_FORMAT_YUV420,
+     {99, 1, 99, 1, 1, 2, 1, 2},
+     {1, 1, 99, 99},
+     true},
 };
 
 using ValidClapPropertyTest = ::testing::TestWithParam<ValidClapPropertyParam>;
 
 INSTANTIATE_TEST_SUITE_P(Parameterized, ValidClapPropertyTest,
                          ::testing::ValuesIn(kValidClapPropertyTestParams));
 
-// Positive tests for the avifCropRectConvertCleanApertureBox() function.
+// Positive tests for the avifCropRectFromCleanApertureBox() function.
 TEST_P(ValidClapPropertyTest, ValidateClapProperty) {
   const ValidClapPropertyParam& param = GetParam();
   avifCropRect crop_rect;
   avifDiagnostics diag;
-  EXPECT_TRUE(avifCropRectConvertCleanApertureBox(&crop_rect, &param.clap,
-                                                  param.width, param.height,
-                                                  param.yuv_format, &diag))
+  ASSERT_TRUE(avifCropRectFromCleanApertureBox(
+      &crop_rect, &param.clap, param.width, param.height, &diag))
       << diag.error;
+  const avifBool upsample_before_cropping =
+      avifCropRectRequiresUpsampling(&crop_rect, param.yuv_format);
   EXPECT_EQ(crop_rect.x, param.expected_crop_rect.x);
   EXPECT_EQ(crop_rect.y, param.expected_crop_rect.y);
   EXPECT_EQ(crop_rect.width, param.expected_crop_rect.width);
   EXPECT_EQ(crop_rect.height, param.expected_crop_rect.height);
+  EXPECT_EQ(upsample_before_cropping, param.expected_upsample_before_cropping);
+
+  // Deprecated function coverage.
```

Comment:
This coverage is insufficient. If `avifCropRectConvertCleanApertureBox()` returns true, we should also check `crop_rect`. Here is one way to do that:

```
  const avifBool success =
    avifCropRectConvertCleanApertureBox(&crop_rect, &param.clap,
                                        param.width, param.height,
                                        param.yuv_format, &diag);
  EXPECT_EQ(success, !upsample_before_cropping);
  if (success) {
    EXPECT_EQ(crop_rect.x, param.expected_crop_rect.x);
    EXPECT_EQ(crop_rect.y, param.expected_crop_rect.y);
    EXPECT_EQ(crop_rect.width, param.expected_crop_rect.width);
    EXPECT_EQ(crop_rect.height, param.expected_crop_rect.height);
  }
```